openshift-ci/README.md

```diff
@@ -0,0 +1,193 @@
+created.
+
+
+## images
+
+There are three container images that are built as part of this job.
+Before this step, a couple of base images are tagged from existing
+published images.  The first one is a CentOS 7 image referred as `os`
+in the CI configuration.  The other one is the operator registry image
+which contains all the binaries to run a gRPC based registry.  The
+operator image registry is referred as `operator-registry` in the CI
+configuration.
+
+### devconsole-operator
+
+Thee CentOS 7 image (`os`) is used as the base image for creating
+`devconsole-operator` image.  The operator binary available inside
+`bin` container image is copied over here.  The Dockerfile used is
+available here: `openshift-ci/Dockerfile.deploy`
+
+The image produced can be pulled from here:
+`registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:devconsole-operator`
+
+### operator-registry-base
+
+The `operator-registry` is used as the base image for creating
+`operator-registry-base` image.  This is an intermediate image used to
+propagate value of `OPENSHIFT_BUILD_NAMESPACE` environment variable to
+`devconsole-operator-registry` image build.  This intermediate image
+is going to be used as the base image for
+`devconsole-operator-registry`.  The original `operator-registry`
+image has a a different value for `OPENSHIFT_BUILD_NAMESPACE`
+environment variable, which must be set when that image was built.
+
+The Dockerfile used is available here:
+`openshift-ci/Dockerfile.registry.intermediate` As you can see, the
+Dockerfile file has only one line with a `FROM` statement, which is
+going to replaced with `operator-registry` image during image build.
+
+### devconsole-operator-registry
+
+The `operator-registry-base` is used as the base image for creating
+`devconsole-operator-registry` image.
+
+The Dockerfile used is available here:
+`openshift-ci/Dockerfile.registry.build`
+
+The image produced can be pulled from here:
+`registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:devconsole-operator-registry`
+
+## tests
+
+### lint
+
+The lint runs the GolangCI Lint, YAML Lint and Operator Courier.
+GolangCI is a Go program, whereas the other two are written in Python.
+So, Python 3 is a perquisite to run lint.
+
+The GolangCI Lint program runs multiple Go lint tools against the
+repository.  GolangCI Lint is runs lint concurrently and complete
+execution in few seconds. But there is one caveat, it requires lots of
+memory.  The memory limit has been increased to 6GB to accommodate the
+requirement.  As of now there is no configuration provided to run
+GolangCI Lint.
+
+The YAML Lint tools validate all the YAML configuration files.  It
+excludes the `vendor` directory while running.  There is a
+configuration file at top-level directory of the source code:
+`.yamllint`.
+
+The Operator Courier checks for the validity of Operator Lifecycle
+Manager (OLM) manifests.  That includes Cluster Service Version (CSV)
```

Comment:
It's good that the doc spells out the meaning of "CSV" - this should help new users.

Reply:
:+1: